vocabulary-draft/source-code-version-control.md

```diff
@@ -0,0 +1,37 @@
+# Source Code Version Control Events
+
+__Note:__ This is a work-in-progress draft version and is being worked on by members of the Events SIG. You are very welcome to join the work and the discussions!
+
+This phase includes events related to changes in Source Code repositories that are relevant from a Continuous Delivery perspective.
+
+## Repository Events
+
+These events are related to Source Code repositories
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+
+
+
+Repository Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `cd.repository**` prefix. For example `cd.repository.created` or `cd.repository.changeapproved`
+- **Repository URL**: indicates the location of the source code repository for API operations, this URL needs to include the protocol used to connect to the repository. For example `git://` , `ssh://`, `https://`
+- **Repository Name**: friendly name to list this repository to users
+
+Optional attributes: 
+- **Repository Owner**: indicates who is the owner of the repository, usually a `user` or an `organization`
+- **Repository View URL**: URL to access the repository from a user web browser
+
+From each repository you can emit events related with proposed source code changes. Each change can include a single or multiple commits that can also be tracked. 
+
+- **Change Created Event**: a source code change was created and submitted to a repository specific branch. Examples: PullRequest sent to Github, MergeRequest sent to Gitlab, Change created in Gerrit
```

Comment:
A **change** is an abstraction created by tools like gerrit, github and gitlab on top of `git`. 
We need to have **change** specific events, but I wonder if we should have events for the underlying repo too.
For instance when a **change** is closed, you could from zero to many commits added to the repo. 
Do we want separate events for that, or do we want to embed that information in the change event? 
You can have commits added to a repo without a change too, as most of this systems allow direct interactions with the underlying source code version control system. 

The data model can easily explode in this space (see all github events for instance), so I think it's fine if we can it simple in the beginning, but I think it's important to make sure we have an event when something changes in the repository.

Reply:
@afrittoli I tend to agree.. let's keep it simple for now, but let's make sure that we discuss how can we tackle more granular changes. We need to have a clear way to make decisions about for example github events, what situations to cover and which ones are not strictly relevant. For making those decisions I propose to define a set of scenarios (and PoCs) that we want to cover with the first iteration and then expand from there.